### PR TITLE
(#1508) Sizing Tokens Utility Page

### LINKED ISSUE #1508: CR: Adopt documentation site Spacing Units page to NCIDS ownership
### Request Description
This change request is to fully adopt spacing unit tokens within the NCIDS from being a pointer to USWDS to our full ownership. This change will require:

- [ ] IA/design updates to the [Spacing Units doc site page](https://designsystem-dev.cancer.gov/develop/foundations/spacing-units) with full instructions on use
- [ ] Design updates to the [community Figma](https://www.figma.com/file/4FeMfprVErO6esR8FLczp5/NCIDS-Design-Kit?type=design&node-id=0-1&mode=design&t=Ec8CKPLUg3f7Jlm4-0)
- [ ] Developer updates to the code to adopt the spacing unit tokens within the NCIDS

### What's the expected change?
- The doc site is updated to show users how to fully use spacing unit tokens without pointing to USWDS
- The community Figma file will be updated to include the spacing unit tokens
- The NCIDS code will be updated to fully adopt the spacing unit tokens

### What's the current functionality?
- The doc site directs users to [USWDS](https://designsystem.digital.gov/design-tokens/spacing-units/) information on how to use spacing unit tokens.

### Related Tickets
* Original story: #1193 
* Related story: #1327

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #1508

## Pull Request Details

_Add description_

Closes #<TICKET-NUMBER>

## Author PR Checklist
Items that the author of the PR is responsible for checking before submitted the PR.

### General:
- [x] I have reviewed the acceptance criteria defined in the ticket and ensured the work has been completed.
- [x] The commit message passes [all quality commit message standards](https://github.com/NCIOCPL/ncids/wiki/Creating-Quality-Commit-Messages).
- [x] Unit tests have been updated or created to reflect any javascript changes.
- [x] Storybook scenarios have been updated or created to reflect any html/css/js changes.

### Accessibility:
- [x] WCAG 2.1 Level AA requirements have been met.

### Development:
- [x] Any new or updated javascript code has 100% unit test coverage.
- [x] New or updated breakpoints have regression images.
- [x] Breaking changes have been thoroughly documented in the PR.

## Product Reviewer PR Checklist
Items the product team is responsible for reviewing.

### General:
- [x] There are no unexpected or unapproved regression image changes. 
- [x] Functionality of interactive elements meet the acceptance criteria.
- [x] The product is visually and functionally the same across the different browsers.

### Accessibility:
- [x] AxeDev Tools: there are no new or outstanding accessibility issues introduced in this PR.
- [x] Lighthouse: scores have not noticeably decreased during this PR.
- [x] Wave: there are no new errors or contrast errors introduced in this PR.

## Design Reviewer PR Checklist
Items the design team is responsible for reviewing.  

### General:
- [x] New or updated features introduced in this PR are developed mobile-first.
- [x] Breakpoint changes and regression images match those breakpoints.
- [x] This PR has been tested in all supported browsers at all breakpoints.

## Developer Reviewer PR Checklist
Items the development team is responsible for reviewing.

### General:
- [ ] New code passes code quality standards set by industry standards.
- [ ] The expected Storybook stories have been added or updated for the new or updated feature.
- [ ] The expected unit tests have been added or updated for the new or updated feature.

### Accessibility:
- [ ] VoiceOver: Described content matches with what was expected.
- [ ] Keyboard navigation: new or updated features and content are navigable via the keyboard.

### CODE REVIEW

**Comment by bryanpizzillo** [comment]
@karlikpj we should be replacing https://designsystem-dev.cancer.gov/pr-1771/foundations/spacing-units/#spacing-units-in-uswds with your tokens. @adrianacastaneda do you agree?

File: docs/content/foundations/spacing/spacing-unit-tokens.mdx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+          - token_name: '-2.5, neg-205'
+            multiple: '-2.5'
+            calculated_value: '-20px'
+          - token_name: '-3 ,neg-3'
+            multiple: '-3'
+            calculated_value: '-24px'
+          - token_name: '-4, neg-4'
+            multiple: '-4'
+            calculated_value: '-32px'
+          - token_name: ', neg-5'
```
Comment (bryanpizzillo) [inline docs/content/foundations/spacing/spacing-unit-tokens.mdx:133]: @karlikpj I assume this is a mistake and USWDS did not exclude `"-5"` as a valid unit.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
We should add the following content to the intro section of the spacing tokens.

```
The documentation below displays a target value in pixels, but the final output for tokens above 2px and below -2px is the eqivalent value in rem, which will vary depending on your project settings.
```

As well as 
```
Note: Some units tokens come in two flavors: a numeric token (like 0.5 or -2) and a string token (like '05' or 'neg-2'). Utility classes use unquoted string tokens exclusively when they exist (like .margin-x-neg-2). Mixins and functions usually accept either, and we will note any exceptions.
```

The note is especially important.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
@karlikpj `'05'` still seems to be off on the display. It is basically the same, or maybe slightly bigger, than `1`. Inspector says the span is 8px high.

File: docs/content/foundations/spacing/spacing-unit-tokens.mdx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+browser_title: NCI Design System Spacing Unit Tokens
+page_title: Spacing Unit Tokens
+nav_label: Spacing Unit Tokens
+nav_order: 10
+template_type: utility
+
+design_tokens:
+  - heading: Spacing Tokens
+    description: |
+      The documentation below displays a target value in pixels, but the final output for tokens above 2px and below -2px is the eqivalent value in rem, which will vary depending on your project settings.
```
Comment (bryanpizzillo) [inline docs/content/foundations/spacing/spacing-unit-tokens.mdx:11]: I asked for the NOTE to go here below this paragraph. 

Add this here:
```
**Note:** Some units tokens come in two flavors: a **numeric token** (like 0.5 or -2) and a **string token** (like '05' or 'neg-2'). Utility classes use unquoted string tokens exclusively when they exist (like .margin-x-neg-2). Mixins and functions usually accept either, and we will note any exceptions.
```

File: docs/content/foundations/spacing/spacing-unit-tokens.mdx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            calculated_value: '-120px'
+---
+
+The NCIDS uses the U.S. Web Design System (USWDS) spacing units out of the box from USWDS. Their spacing unit tokens are based on multiples of 8px, with additional tokens for small sizes, named tokens for large sizes, and a more limited selection of negative tokens.
+
+## Applying Spacing Units for Designers
+
+For more on how to apply spacing units in Figma, refer to our [Figma Guide](/get-started/figma-guide).
+
+**Note:** Some units tokens come in two flavors: a **numeric token** (like 0.5 or -2) and a **string token** (like '05' or 'neg-2'). Utility classes use unquoted string tokens exclusively when they exist (like .margin-x-neg-2). Mixins and functions usually accept either, and we will note any exceptions.
```
Comment (bryanpizzillo) [inline docs/content/foundations/spacing/spacing-unit-tokens.mdx:163]: Remove this here once added to the top.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]

File: docs/content/foundations/spacing/spacing-unit-tokens.mdx (hunk 2 above)
Comment (karlikpj) [reply docs/content/foundations/spacing/spacing-unit-tokens.mdx:11]: the note content breaks the utility page when placed in the description / doesn't handle markdown.
```YAMLException: unidentified alias "*Note:**" at line 12, column 10:
      **Note:** Some units tokens come in two f ... ```
      
We can move out of description and place above the content listed below to get the correct order, as the token yaml only has one description per object.

**Comment by bryanpizzillo** [comment]
Note about the wonky markup for the note:
The BRs below are because markdown in HTML tags are not working, and we need to separate these two blocks. An empty `<p></p>` tag is getting processed as a `<p>` tag wrapping the note. I also don't want to make the note all HTML. A `>` quote does seem to work ok ish, but we have no styling for `<blockquote>`, so it goes all wonky.

**Review by bryanpizzillo — APPROVED** [review]

**Review by bennettcc — APPROVED** [review]

**Comment by bennettcc** [comment]
Andy and Sarina approved on 7/10 - adding the product checkmark

**Pull request merged**